core/src/main/java/org/alfresco/util/TempFileProvider.java

```diff
@@ -468,7 +521,32 @@ private static int removeFiles(File directory, long removeBefore, long longLifeB
                         {
                             logger.debug("Deleting temp file: " + file);
                         }
-                        file.delete();
+                        // only delete if the limits allow
+                        if (maxFilesToDelete != null && maxFilesToDelete.get() <= 0 ||
+                            maxTimeToRun != null && ((jobStartTime + maxTimeToRun.toMillis()) < System.currentTimeMillis()))
+                        {
+                            return count;
+                        }
+                        else
```

Comment:
No need to use `else` after a `if { return/break/continue }` statement. It's redundand and needlessly  expands the code horizontally.